Content/Console/build.fsx

```diff
@@ -161,12 +226,40 @@ let clean _ =
     ]
     |> Seq.iter Shell.rm
 
+let updateChangelog ctx =
+    let description, unreleasedChanges =
+        match changelog.Unreleased with
+        | None -> None, []
+        | Some u -> u.Description, u.Changes
+    let verStr = ctx |> getVersionNumber "RELEASE_VERSION"
+    let newVersion = SemVer.parse verStr
+    if changelog.Entries |> List.exists (fun entry -> entry.SemVer = newVersion) then
+        let entry = changelog.Entries |> List.find (fun entry -> entry.SemVer = newVersion)
+        Trace.traceErrorfn "Version %s already exists in %s, released on %s" verStr changelogFilename (if entry.Date.IsSome then entry.Date.Value.ToString("yyyy-MM-dd") else "(no date specified)")
+        failwith "Can't release with a duplicate version number"
+    if changelog.Entries |> List.exists (fun entry -> entry.SemVer > newVersion) then
+        let entry = changelog.Entries |> List.find (fun entry -> entry.SemVer > newVersion)
+        Trace.traceErrorfn "You're trying to release version %s, but a later version %s already exists, released on %s" verStr entry.SemVer.AsString (if entry.Date.IsSome then entry.Date.Value.ToString("yyyy-MM-dd") else "(no date specified)")
+        failwith "Can't release with a version number older than an existing release"
+    let versionTuple version = (version.Major, version.Minor, version.Patch)
+    let prereleaseEntries = changelog.Entries |> List.filter (fun entry -> entry.SemVer.PreRelease.IsSome && versionTuple entry.SemVer = versionTuple newVersion)
+    let prereleaseChanges = prereleaseEntries |> List.collect (fun entry -> entry.Changes |> List.filter (not << isEmptyChange))
+    let assemblyVersion, nugetVersion = Changelog.parseVersions newVersion.AsString
+    linkReferenceForLatestEntry <- mkLinkReference newVersion changelog
+    let newEntry = Changelog.ChangelogEntry.New(assemblyVersion.Value, nugetVersion.Value, Some System.DateTime.Today, description, unreleasedChanges @ prereleaseChanges, false)
+    let newChangelog = Changelog.Changelog.New(changelog.Header, changelog.Description, None, newEntry :: changelog.Entries)
+    latestEntry <- newEntry
+    newChangelog
+    |> Changelog.save changelogFilename
+    // Changelog.save doesn't write a final newline, so we add one when writing out the new link reference
+    sprintf "\n%s\n" linkReferenceForLatestEntry |> File.writeString true changelogFilename
+
 let dotnetRestore _ =
     [sln]
     |> Seq.map(fun dir -> fun () ->
         let args =
             [
-                sprintf "/p:PackageVersion=%s" releaseNotes.NugetVersion
+                sprintf "/p:PackageVersion=%s" latestEntry.NuGetVersion
```

Comment:
[Based on my findings](https://github.com/TheAngryByrd/MiniScaffold/pull/175#issuecomment-579547575), I think we can eliminate this line.

Reply:
Agreed, and I'll also change the UpdateChangelog target to come after DotnetRestore and before GenerateAssemblyInfo; that's a more natural place for it anyway.

Reply:
Done in dbc1a55.